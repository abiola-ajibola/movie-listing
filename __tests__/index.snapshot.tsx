import React from "react";
import renderer from "react-test-renderer";
import Index from "../pages/index";

const imageBaseUrl = "https://image.tmdb.org/t/p/w300";
const apiData = [
  {
    adult: false,
    backdrop_path: "/1EAxNqdkVnp48a7NUuNBHGflowM.jpg",
    genre_ids: [16, 28, 18, 878],
    id: 283566,
    original_language: "ja",
    original_title: "シン・エヴァンゲリオン劇場版:||",
    overview:
      "In the aftermath of the Fourth Impact, stranded without their Evangelions, Shinji, Asuka, and Rei find refuge in one of the rare pockets of humanity that still exist on the ruined planet Earth. There, each of them live a life far different from their days as an Evangelion pilot. However, the danger to the world is far from over. A new impact is looming on the horizon—one that will prove to be the true end of Evangelion.",
    popularity: 286.466,
    poster_path: "/jDwZavHo99JtGsCyRzp4epeeBHx.jpg",
    release_date: "2021-03-08",
    title: "Evangelion: 3.0+1.0 Thrice Upon a Time",
    video: false,
    vote_average: 8.7,
    vote_count: 256,
  },
  {
    adult: false,
    backdrop_path: "/9Xw0I5RV2ZqNLpul6lXKoviYg55.jpg",
    genre_ids: [18, 80],
    id: 278,
    original_language: "en",
    original_title: "The Shawshank Redemption",
    overview:
      "Framed in the 1940s for the double murder of his wife and her lover, upstanding banker Andy Dufresne begins a new life at the Shawshank prison, where he puts his accounting skills to work for an amoral warden. During his long stretch in prison, Dufresne comes to be admired by the other inmates -- including an older prisoner named Red -- for his integrity and unquenchable sense of hope.",
    popularity: 63.552,
    poster_path: "/q6y0Go1tsGEsmtFryDOJo3dEmqu.jpg",
    release_date: "1994-09-23",
    title: "The Shawshank Redemption",
    video: false,
    vote_average: 8.7,
    vote_count: 19561,
  },
  {
    adult: false,
    backdrop_path: "/jtAI6OJIWLWiRItNSZoWjrsUtmi.jpg",
    genre_ids: [10749],
    id: 724089,
    original_language: "en",
    original_title: "Gabriel's Inferno Part II",
    overview:
      "Professor Gabriel Emerson finally learns the truth about Julia Mitchell's identity, but his realization comes a moment too late. Julia is done waiting for the well-respected Dante specialist to remember her and wants nothing more to do with him. Can Gabriel win back her heart before she finds love in another's arms?",
    popularity: 8.003,
    poster_path: "/x5o8cLZfEXMoZczTYWLrUo1P7UJ.jpg",
    release_date: "2020-07-31",
    title: "Gabriel's Inferno Part II",
    video: false,
    vote_average: 8.7,
    vote_count: 1297,
  },
];

it("renders homepage unchanged", () => {
  const tree = renderer
    .create(
      <Index
        imageBaseUrl={imageBaseUrl}
        apiData={apiData}
        meta={{ page: 1, total_pages: 1, total_results: apiData.length }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
