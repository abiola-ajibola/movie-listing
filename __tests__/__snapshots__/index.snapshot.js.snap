// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div
  className="container"
>
  <main>
    <h1
      className="title"
    >
      Welcome to My Movies List
    </h1>
    <p
      className="description"
      data-testid="subtitle"
    >
      Scroll down to see the top 500 movies
    </p>
    <div
      className="switcherWrapper"
    >
      <div>
        <span
          style={
            Object {
              "margin": "0 0 5px",
              "textAlign": "center",
            }
          }
        >
          Sort: 
        </span>
        <input
          className="checkbox"
          defaultChecked={true}
          id="ascending"
          name="switch"
          onChange={[Function]}
          type="radio"
          value="ascending"
        />
        <label
          aria-label="ascending"
          htmlFor="ascending"
        >
          ↑
        </label>
        <input
          className="checkbox"
          id="descending"
          name="switch"
          onChange={[Function]}
          type="radio"
          value="descending"
        />
        <label
          aria-label="descending"
          htmlFor="descending"
        >
          ↓
        </label>
      </div>
    </div>
    <div
      className="cardWrapper"
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <div
        className="starWrapper"
      >
        <span
          className="star"
          onClick={[Function]}
          style={
            Object {
              "color": "black",
            }
          }
        >
          ★
        </span>
      </div>
      <a
        href="https://www.themoviedb.org/movie/283566"
      >
        <div
          className="card"
        >
          <div
            className="imageWrapper"
          >
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "display": "block",
                    "maxWidth": "100%",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjE2OSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                  style={
                    Object {
                      "border": "none",
                      "display": "block",
                      "margin": 0,
                      "maxWidth": "100%",
                      "padding": 0,
                    }
                  }
                />
              </div>
              <img
                alt="Evangelion: 3.0+1.0 Thrice Upon a Time"
                className="image"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="Evangelion: 3.0+1.0 Thrice Upon a Time"
                  className="image"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2F1EAxNqdkVnp48a7NUuNBHGflowM.jpg&w=640&q=75"
                  srcSet="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2F1EAxNqdkVnp48a7NUuNBHGflowM.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2F1EAxNqdkVnp48a7NUuNBHGflowM.jpg&w=640&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </div>
          </div>
          <div
            className="detailsWrapper"
          >
            <ul
              className="cardItemsWrapper"
            >
              <li>
                <h3
                  className="title"
                  style={
                    Object {
                      "color": "darkgray",
                    }
                  }
                >
                  Title
                </h3>
                <p
                  className="value"
                  data-testid="title-text"
                  style={
                    Object {
                      "color": "black",
                    }
                  }
                >
                  Evangelion: 3.0+1.0 Thrice Upon a Time
                </p>
              </li>
              <li>
                <h3
                  className="title"
                  style={
                    Object {
                      "color": "darkgray",
                    }
                  }
                >
                  Current Rating
                </h3>
                <p
                  className="value"
                  style={
                    Object {
                      "color": "black",
                    }
                  }
                >
                  8.7
                </p>
              </li>
              <li>
                <h3
                  className="title"
                  style={
                    Object {
                      "color": "darkgray",
                    }
                  }
                >
                  Year
                </h3>
                <p
                  className="value"
                  style={
                    Object {
                      "color": "black",
                    }
                  }
                >
                  2021
                </p>
              </li>
            </ul>
          </div>
        </div>
      </a>
    </div>
    <div
      className="cardWrapper"
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <div
        className="starWrapper"
      >
        <span
          className="star"
          onClick={[Function]}
          style={
            Object {
              "color": "black",
            }
          }
        >
          ★
        </span>
      </div>
      <a
        href="https://www.themoviedb.org/movie/278"
      >
        <div
          className="card"
        >
          <div
            className="imageWrapper"
          >
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "display": "block",
                    "maxWidth": "100%",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjE2OSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                  style={
                    Object {
                      "border": "none",
                      "display": "block",
                      "margin": 0,
                      "maxWidth": "100%",
                      "padding": 0,
                    }
                  }
                />
              </div>
              <img
                alt="The Shawshank Redemption"
                className="image"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="The Shawshank Redemption"
                  className="image"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2F9Xw0I5RV2ZqNLpul6lXKoviYg55.jpg&w=640&q=75"
                  srcSet="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2F9Xw0I5RV2ZqNLpul6lXKoviYg55.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2F9Xw0I5RV2ZqNLpul6lXKoviYg55.jpg&w=640&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </div>
          </div>
          <div
            className="detailsWrapper"
          >
            <ul
              className="cardItemsWrapper"
            >
              <li>
                <h3
                  className="title"
                  style={
                    Object {
                      "color": "darkgray",
                    }
                  }
                >
                  Title
                </h3>
                <p
                  className="value"
                  data-testid="title-text"
                  style={
                    Object {
                      "color": "black",
                    }
                  }
                >
                  The Shawshank Redemption
                </p>
              </li>
              <li>
                <h3
                  className="title"
                  style={
                    Object {
                      "color": "darkgray",
                    }
                  }
                >
                  Current Rating
                </h3>
                <p
                  className="value"
                  style={
                    Object {
                      "color": "black",
                    }
                  }
                >
                  8.7
                </p>
              </li>
              <li>
                <h3
                  className="title"
                  style={
                    Object {
                      "color": "darkgray",
                    }
                  }
                >
                  Year
                </h3>
                <p
                  className="value"
                  style={
                    Object {
                      "color": "black",
                    }
                  }
                >
                  1994
                </p>
              </li>
            </ul>
          </div>
        </div>
      </a>
    </div>
    <div
      className="cardWrapper"
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <div
        className="starWrapper"
      >
        <span
          className="star"
          onClick={[Function]}
          style={
            Object {
              "color": "black",
            }
          }
        >
          ★
        </span>
      </div>
      <a
        href="https://www.themoviedb.org/movie/724089"
      >
        <div
          className="card"
        >
          <div
            className="imageWrapper"
          >
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "display": "block",
                    "maxWidth": "100%",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjE2OSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                  style={
                    Object {
                      "border": "none",
                      "display": "block",
                      "margin": 0,
                      "maxWidth": "100%",
                      "padding": 0,
                    }
                  }
                />
              </div>
              <img
                alt="Gabriel's Inferno Part II"
                className="image"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="Gabriel's Inferno Part II"
                  className="image"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2FjtAI6OJIWLWiRItNSZoWjrsUtmi.jpg&w=640&q=75"
                  srcSet="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2FjtAI6OJIWLWiRItNSZoWjrsUtmi.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2FjtAI6OJIWLWiRItNSZoWjrsUtmi.jpg&w=640&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </div>
          </div>
          <div
            className="detailsWrapper"
          >
            <ul
              className="cardItemsWrapper"
            >
              <li>
                <h3
                  className="title"
                  style={
                    Object {
                      "color": "darkgray",
                    }
                  }
                >
                  Title
                </h3>
                <p
                  className="value"
                  data-testid="title-text"
                  style={
                    Object {
                      "color": "black",
                    }
                  }
                >
                  Gabriel's Inferno Part II
                </p>
              </li>
              <li>
                <h3
                  className="title"
                  style={
                    Object {
                      "color": "darkgray",
                    }
                  }
                >
                  Current Rating
                </h3>
                <p
                  className="value"
                  style={
                    Object {
                      "color": "black",
                    }
                  }
                >
                  8.7
                </p>
              </li>
              <li>
                <h3
                  className="title"
                  style={
                    Object {
                      "color": "darkgray",
                    }
                  }
                >
                  Year
                </h3>
                <p
                  className="value"
                  style={
                    Object {
                      "color": "black",
                    }
                  }
                >
                  2020
                </p>
              </li>
            </ul>
          </div>
        </div>
      </a>
    </div>
  </main>
</div>
`;
